src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,197 @@
+                                           :onboarding?           true
+                                           :syncing?              syncing?}]
+                             :on-fail    [:onboarding/biometrics-fail]}]}]])
+
+         (not enable-biometrics?)
+         (conj [:dispatch
+                [:onboarding/notifications-setup
+                 {:biometrics?           false
+                  :biometrics-supported? true
+                  :onboarding?           true
+                  :syncing?              syncing?}]]))})
+
+(rf/reg-event-fx :onboarding/biometrics-setup-start biometrics-setup-start)
+
+(defn biometrics-setup-done
+  [{:keys [db]} [{:keys [on-success on-fail]} {:keys [error]}]]
+  {:db (assoc-in db [:onboarding/profile :auth-method] constants/auth-method-biometric)
+   :fx (cond-> []
+         (some? error)
+         (conj [:dispatch (conj on-fail error)])
+         :else
+         (conj [:dispatch on-success]))})
+
+(rf/reg-event-fx :onboarding/biometrics-setup-done biometrics-setup-done)
+
 (rf/reg-event-fx
- :onboarding/finish-onboarding
- (fn [_ [notifications-enabled?]]
-   {:fx [(when notifications-enabled?
-           [:dispatch [:push-notifications/switch {:notifications-enabled true}]])
-         [:dispatch [:shell/change-tab shell.constants/default-selected-stack]]
-         [:dispatch [:update-theme-and-init-root :screen/shell-stack]]
-         [:dispatch [:profile/toggle-testnet-mode-banner]]
-         [:dispatch [:universal-links/process-stored-event]]]}))
+ :onboarding/biometrics-fail
+ (fn [_ [error]]
+   {:dispatch [:biometric/show-message (ex-cause error)]}))
 
-(rf/defn enable-biometrics
-  {:events [:onboarding/enable-biometrics]}
-  [_]
+(defn enable-biometrics
+  [_ [{:keys [on-done]}]]
   {:fx [[:dispatch
          [:biometric/authenticate
-          {:on-success #(rf/dispatch [:onboarding/biometrics-done])
-           :on-fail    #(rf/dispatch [:onboarding/biometrics-fail %])}]]]})
+          {:on-success #(rf/dispatch on-done)
+           :on-fail    #(rf/dispatch (conj on-done %))}]]]})
+
+(rf/reg-event-fx :onboarding/enable-biometrics enable-biometrics)
+
+;;; -------
+
+(defn inject-local-profile-storage
+  [context]
+  (let [db            (interceptor/get-coeffect context :db)
+        key-uid       (get-in db [:profile/profile :key-uid])
+        local-profile (mmkv/get-object key-uid)]
+    (assoc-in context
+     [:coeffects :local-profile-storage]
+     local-profile)))
+
+(def local-profile-storage-interceptor
+  (interceptor/->interceptor
+   :id     :local-profile-storage-interceptor
+   :before inject-local-profile-storage))
+
+(rf/reg-fx :effects.profile/save-notifications-prompted
+ (fn [{:keys [key-uid]}]
+   (let [profile-storage (mmkv/get-object key-uid {})]
+     (mmkv/set-object key-uid
+                      (assoc profile-storage :notifications-prompted? true)))))
+
+(rf/reg-fx :effects.profile/remove-local-profile-storage
```

Comment:
For maintenance purposes, I think that mmkv manipulations should be decoupled from specific Status bounded contexts. So we could have an `:effects.kv/delete` effect instead.

In future PRs, it would also be good to consider abstracting away the `mmkv` lib under a stable interface for any local and synchronous kv store. Ideally it should've been done from the initial integration of mmkv, but IIRC time wasn't on our side.